sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,235 @@
+
+class EventHubProducerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        *,
+        event_hub_path: str = None,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        auth_timeout: float = 60,
+        user_agent: str = None,
+        retry_total: int = 3,
+        transport_type=None) -> EventHubProducerClient:
```

Comment:
My gut feels like this is a naive question, but: Is there a reason we haven't defined the transport_type type?

Reply:
So good a question.
@annatisch we expose azure.eventhub.TransportType but the actual implementation is 
uamqp.constants.TransportType.

Reply:
Yeah good point - we could either:
- Define our own transport type in EH
- Expose the uamqp transport type in EH __init__.py

Reply:
A Python-specific question - so wont raise in the API review.